### DISCUSSION TITLE
Revivers / Parsing Array and JSONb properties in Supabase Realtime

### DISCUSSION THREAD

**Discussion by stevenolay** [supabase/supabase#3683] [Questions] [answered]
This is an example record that I get from the Supabase realtime socket. 

```
{
   "active":"f",
   "nickname":null,
   "phone_numbers":"{+165055511111, +16505552222}",
   "photo_url":null,
   "primary_email":null,
   "primary_email_verified":null,
   "primary_phone_number":"+165055511111",
}
```

Booleans remain as strings in the form "f" or "t"

Arrays come back in the following format "{val1, val2, val3}"

Are there any built in utilities/utilities you can point me to to transform these properties into native JSON arrays and booleans?

I.E

```
{
   "active": false,
   "nickname":null,
   "phone_numbers":["+165055511111", "+16505552222"],
   "photo_url":null,
   "primary_email":null,
   "primary_email_verified":null,
   "primary_phone_number":"+165055511111",
}
```

Am i potentially missing a setting in the setup of the realtime server or my socket subscription?

**Comment by GaryAustin1** [comment, marked as answer]
For what it is worth, in Supabase's realtime.js they have a transformer function to convert all the responses coming back from the channel into json before passing it on as a payload response.

**Reply by stevenolay** [reply]
Ahh, Are you referring to this?

https://github.com/supabase/realtime-js/blob/master/src/lib/transformers.ts

I will see if i can clone this and fit it to transform the data in my backend. I figured there had to be some parser somewhere 😅

**Reply by GaryAustin1** [reply]
Yeah, They happened to have fixed something in it a few days ago (yikes?) and trying to keep up on latest JS stability as I get closer to production.

**Reply by stevenolay** [reply]
Ooooo. Okay big yikes hahaha.

Definitely can't put complete trust in the transformer for now. Going to definitely keep an eye out and monitor it.

**Comment by w3b6x9** [comment]
@stevenolay looks like you might be spinning up the Realtime server and/or using a Realtime client directly?

We actually transform Realtime messages in our Supabase client (e.g. `supabase-js`). Have a look at https://github.com/supabase/supabase-js/blob/48a408232600906202bb95793ab71ecb977fc16b/src/lib/SupabaseRealtimeClient.ts#L31 and https://github.com/supabase/supabase-js/blob/48a408232600906202bb95793ab71ecb977fc16b/src/lib/SupabaseRealtimeClient.ts#L35 for reference.

Existing transformers shouldn't have an issue converting your array and boolean data as you've presented here.

**Reply by stevenolay** [reply]
Thanks!

I was able to find my way there thanks to the suggestions of @GaryAustin1 .

All worked perfectly for me so far!
